src/test/scala/examples/RerunWithoutElaboratonAndCompileSpec.scala

```diff
@@ -0,0 +1,51 @@
+// See LICENSE for license details.
+
+package examples
+
+import chisel3._
+import chisel3.iotesters.PeekPokeTester
+import org.scalatest.{FreeSpec, Matchers}
+
+/**
+  * This provides an example of how to re-run a verilator compiled simulation without re-running verilator
+  * Currently this method does re-elaborate the circuit
+  */
+class RerunWithoutElaboratonAndCompileSpec extends FreeSpec with Matchers {
```

Comment:
This capability it part of the existing API.  I stuck this in here because someone was asking about it.  It should probably be expanded upon in documentation.  There should probably be a whole lot more docs on how to set up custom test harnesses and toolchains generally.